### PR TITLE
Added YAML support.

### PR DESCRIPTION
**Pull request by abayer**
Note that right now this still means YAML inside a Groovy string, but
once markerfiles are more flexible...

### CODE REVIEW

**Comment by abayer** [comment]
Closing this because I'm nuking my repo and recreating it.

File: src/main/groovy/org/jenkinsci/plugins/plumber/model/Reporter.groovy [hunk 1]
```diff
@@ -46,6 +46,17 @@ public class Reporter extends AbstractPlumberModel {
 
     }
 
+    public Reporter(Map<String,Object> args) {
+        if (args != null) {
+            validateMapFields(args.keySet())
+
+            this.name = args.name
+            if (args.containsKey("config") && args.config instanceof Map) {
+                this.config = new MappedClosure((Map<String, Object>) args.config)
```
Comment (KostyaSha) [inline src/main/groovy/org/jenkinsci/plugins/plumber/model/Reporter.groovy]: @abayer why do you using `this` everywhere? Ideally you shouldn't use overlapped variable names at all.

Comment (abayer) [inline src/main/groovy/org/jenkinsci/plugins/plumber/model/Reporter.groovy]: Habit when I'm in a constructor, I guess? Looking at the code, the `this` does seem to be unneeded.

Comment (KostyaSha) [inline src/main/groovy/org/jenkinsci/plugins/plumber/model/Reporter.groovy]: Useless this enforces to think more and keep feeling that there is real var somewhere...

Comment (abayer) [inline src/main/groovy/org/jenkinsci/plugins/plumber/model/Reporter.groovy]: Fair enough - I'll fix it!

**Pull request closed without merging**